firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -964,4 +986,154 @@ public void testMultipleUpdatesWhileOffline() {
     QuerySnapshot snapshot2 = waitFor(collection.get(Source.CACHE));
     assertEquals(asList(map("foo", "zzyzx", "bar", "2")), querySnapshotToValues(snapshot2));
   }
+
+  // TODO(orquery): Enable this test when prod supports OR queries.
+  @Ignore
+  @Test
+  public void testOrQueries() {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "doc1", map("a", 1, "b", 0),
+            "doc2", map("a", 2, "b", 1),
+            "doc3", map("a", 3, "b", 2),
+            "doc4", map("a", 1, "b", 3),
+            "doc5", map("a", 1, "b", 1));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+
+    // Two equalities: a==1 || b==1.
```

Comment:
Can we turn some of the cases here a mix of equality and inequality?

Reply:
There are a few test cases with the mix of equality and inequality below.
a>2 || b==1
(a==1) || (b > 0) LIMIT 2
(a==1) || (b > 0) LIMIT_TO_LAST 2

Since adding an inequality automatically adds an OrderBy, we do want to have some tests without any inequalities.